Marlin/src/module/temperature.cpp

```diff
@@ -54,32 +86,62 @@
   #endif
   #ifndef MAX31865_MOSI_PIN
     #define MAX31865_MOSI_PIN MOSI_PIN
-  #endif
-  #ifndef MAX31865_MISO_PIN
+  #endif  
+  #if !defined(MAX31865_MISO_PIN) && PIN_EXISTS(MAX6675_DO)
     #define MAX31865_MISO_PIN MAX6675_DO_PIN
   #endif
-  #ifndef MAX31865_SCK_PIN
+  #if !defined(MAX31865_SCK_PIN) && PIN_EXISTS(MAX6675_SCK)
     #define MAX31865_SCK_PIN  MAX6675_SCK_PIN
   #endif
   #if MAX6675_0_IS_MAX31865 && PIN_EXISTS(MAX31865_CS)
     #define HAS_MAX31865 1
     Adafruit_MAX31865 max31865_0 = Adafruit_MAX31865(MAX31865_CS_PIN
-      #if MAX31865_CS_PIN != MAX6675_SS_PIN
```

Comment:
The original code, If I'm reading right, just means: "are the two sensor the same?", because if the have the same CS/SS pin, they are in the same spi port and are, in fact, the same device...

This CS/SS looks very confusing.

Reply:
The problem arose here:
```
#define MAX6675_DO_PIN                    P1_01 
#define MAX6675_SCK_PIN                   P1_14
#define MAX31865_MOSI_PIN                 P1_15
#define MAX6675_SS_PIN                    TEMP_0_PIN 
```
The Marlin user would think that is will lead to Software SPI because you have define the DO_PIN and SCK_PIN and MOSI_PIN along with the SS_PIN. But, because of the code preceding the line number 67, equates the MAX6675_SS_PIN to MAX31865_CS_PIN for the object's instantiation call.  The use ends up going into Hardware SPI call instead of the Software SPI.  So in actuality the MAX31865_CS_PIN != MAX6675_SS_PIN keeps the user from going to Software SPI when that is really what he wanted.  Now if the user would have done the following:
```
#define MAX6675_DO_PIN                    P1_01 
#define MAX6675_SCK_PIN                   P1_14
#define MAX31865_MOSI_PIN                 P1_15
#define MAX31865_CS_PIN                    TEMP_0_PIN 
```
Now Software SPI will be used.  I just wan the names to all match so the confusion does not occur.  But if the user already has this setup in his pins file and want to update Marlin to the current branch I do not think we should break his setup that he already has gotten working.  It does not hurt anything to have additional definitions.

Reply:
Changed the name to `MAX31865_USES_SW_SPI` because that is all the `MAX31865_CS_PIN != MAX6675_SS_PIN`  was doing. `MAX31865_CS_PIN != MAX6675_SS_PIN`  use to decide which variables where included into the parameter list for the call to the CLASS for object instantiation.  Fixed this item.